Refactor/todo item

beyondbytes13309 opened the pull request:
### Refactor App.jsx and ToDoItem.jsx to support checkbox state

- Replaced plain todo items with checkbox and label tags
- Added ability to check/uncheck items
- Added a proper delete button for each todo
- Refactored todo item structure { text, checked} 
- Made minor css improvements like spacing
- Used MUI delete icon instead of plain delete button
- Styled checkbox

<img width="506" height="664" alt="pr" src="https://github.com/user-attachments/assets/ff351015-1155-4ed4-8e4f-45fd5b84a9ec" />

beyondbytes13309 committed:
refactor ToDoItem to support checkbox state, add checkItem handler to App component

- updated App to store todo items as objects with text and checked property
- added checkItem function to update checked property of todos
- modified ToDoItem to render checkbox input, label and handle onChange event

beyondbytes13309 committed:
Add delete button to ToDoItem and improve label styling

- passed deleteItem handler to ToDoItem from App
- made todo text label unselectable for users

beyondbytes13309 commented:
Hi @abhaymani421 Can you please take a look at my PR? It's my first one so it will be appreciated! Thanks

abhaymani421 commented:
Thank you for contributing. The added functionality is working alright but the styling is inconsistent especially with the alignment of the tasks and the Delete buttons. for delete button you can use a icon from material UI. The tasks should be left aligned and the delete button should be right aligned making the UI consistent.

beyondbytes13309 committed:
Refactor ToDoItem layout and improved ui

- replaced plain delete button with a MUI icon
- styled checkbox and delete icon
- applied styles to improve spacing

beyondbytes13309 commented:
@abhaymani421 thanks for the feedback. ive done as you said
